session/session.go

```diff
@@ -1494,14 +1502,18 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 	if err := executor.ResetContextOfStmt(s, stmtNode); err != nil {
 		return nil, err
 	}
+	normalizedSQL, digest := s.sessionVars.StmtCtx.SQLDigest()
+	if variable.TopSQLEnabled() {
+		topsql.AttachSQLInfo(ctx, normalizedSQL, digest.Bytes(), "", nil)
```

Comment:
I see that `digest.Bytes()` is frequently used. How about simply accepting `*parser.Digest` in `AttachSQLInfo`?

Reply:
great, done.